docs/preflight.md

```diff
@@ -0,0 +1,12 @@
+## Preflight Checklist
+
+This checklist is to be ran prior to cutting the release branch.
+
+- [ ] Bump GoLang Vendor Dependencies
```

Comment:
Save security fixes, I'd suggest bumping dependencies as the first commit in the new line of development post release release, or on a regular cadence (but not too close to any major/minor release) instead of as one of the last commits to go into a major/minor release.

Why do I suggest this?  Bumping dependencies without a specific reason introduces end user risk for almost no end user benefit.  When we bump early or regularly in a release's lifecycle, we have the entire rest of the cycle to notice issues that may be caused from dependency upgrades.  When done at the end of a cycle cycle, we artificially limit the amount of time/testing to notice issues.

Reply:
:100: